map_view/display-layer-view-state/src/main/java/com/esri/samples/display_layer_view_state/DisplayLayerViewStateSample.java

```diff
@@ -72,82 +69,148 @@ public void start(Stage stage) {
 
-      worldTimeZonesLabel.getStyleClass().add("panel-label");
-      censusLabel.getStyleClass().add("panel-label");
-      facilitiesLabel.getStyleClass().add("panel-label");
+      // create a listener that fires every time a layer's view status changes
+      mapView.addLayerViewStateChangedListener(statusChangeEvent -> {
 
-      // add labels to the control panel
-      controlsVBox.getChildren().addAll(worldTimeZonesLabel, censusLabel, facilitiesLabel);
-
-      // create three layers to add to the ArcGISMap
-      final ArcGISTiledLayer tiledLayer = new ArcGISTiledLayer(SERVICE_TIME_ZONES);
+        // get the layer whose state has changed
+        Layer layer = statusChangeEvent.getLayer();
+        // only update the status if the layer is the feature layer being tracked
+        if (layer != featureLayer) {
+          return;
+        }
+        // get the layer's view status and display the status
+        EnumSet<LayerViewStatus> layerViewStatus = statusChangeEvent.getLayerViewStatus();
+        displayViewStateText(layerViewStatus);
+
+        // if there is an error or warning, format and display the message as an alert
+        ArcGISRuntimeException error = statusChangeEvent.getError();
+        if (error != null) {
+          Throwable cause = error.getCause();
+          String message = (cause != null) ? cause.toString() : error.toString();
+          Alert alert = new Alert(Alert.AlertType.ERROR, message);
+          alert.show();
+        }
+      });
 
-      // this layer will be OUT_OF_SCALE outside of its set min/max scales
-      final ArcGISMapImageLayer imageLayer = new ArcGISMapImageLayer(SERVICE_CENSUS);
-      imageLayer.setMinScale(MIN_SCALE);
-      imageLayer.setMaxScale(MIN_SCALE / 10);
+      // create buttons to toggle the visibility of the feature layer
+      loadLayerButton = new Button("Load Layer");
+      loadLayerButton.getStyleClass().add("panel-button");
+      hideLayerButton = new Button("Hide Layer");
+      hideLayerButton.getStyleClass().add("panel-button");
+      // initially add load layer button to the control panel
+      controlsVBox.getChildren().add(loadLayerButton);
+
+      // create a listener for clicks on the load layer button
+      loadLayerButton.setOnAction(event -> {
+
+        // if the feature layer already exists and is hidden, toggle it's visibility to visible
+        if (featureLayer != null && !featureLayer.isVisible()) {
+          featureLayer.setVisible(true);
+        } else {
+          // create a feature layer from a portal item
+          final PortalItem portalItem = new PortalItem(new Portal("https://runtime.maps.arcgis.com/"),
+            "b8f4033069f141729ffb298b7418b653");
+          featureLayer = new FeatureLayer(portalItem, 0);
+          //  set a minimum and a maximum scale for the visibility of the feature layer
+          featureLayer.setMinScale(40000000);
+          featureLayer.setMaxScale(40000000 / 10);
```

Comment:
The IDE is flagging this up, so looks like it's best to write the number without division. Should be easy enough to tell there's a factor of 10 less here, as the lines are otherwise aligned.
```suggestion
          featureLayer.setMaxScale(4000000);
```